src/Core/Grid/Definition/Factory/ImageTypeGridDefinitionFactory.php

```diff
@@ -0,0 +1,308 @@
+                        'required' => false,
+                    ])
+            )
+            ->add(
+                (new Filter('categories', YesAndNoChoiceType::class))
+                    ->setAssociatedColumn('categories')
+                    ->setTypeOptions([
+                        'required' => false,
+                    ])
+            )
+            ->add(
+                (new Filter('manufacturers', YesAndNoChoiceType::class))
+                    ->setAssociatedColumn('manufacturers')
+                    ->setTypeOptions([
+                        'required' => false,
+                    ])
+            )
+            ->add(
+                (new Filter('suppliers', YesAndNoChoiceType::class))
+                    ->setAssociatedColumn('suppliers')
+                    ->setTypeOptions([
+                        'required' => false,
+                    ])
+            )
+            ->add(
+                (new Filter('stores', YesAndNoChoiceType::class))
+                    ->setAssociatedColumn('stores')
+                    ->setTypeOptions([
+                        'required' => false,
+                    ])
+            )
+            ->add(
+                (new Filter('actions', SearchAndResetType::class))
+                    ->setAssociatedColumn('actions')
+                    ->setTypeOptions([
+                        'reset_route' => 'admin_common_reset_search_by_filter_id',
+                        'reset_route_params' => [
+                            'filterId' => self::GRID_ID,
+                        ],
+                        'redirect_route' => 'admin_image_settings_index',
+                    ])
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getBulkActions(): BulkActionCollection
+    {
+        $bulkActions = new BulkActionCollection();
+        $bulkActions
+            ->add(
+                $this->buildBulkDeleteAction('admin_image_settings_bulk_delete')
+            );
+
+        return $bulkActions;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getGridActions()
+    {
+        return (new GridActionCollection())
+            ->add(
+                (new SimpleGridAction('common_refresh_list'))
+                    ->setName($this->trans('Refresh list', [], 'Admin.Advparameters.Feature'))
+                    ->setIcon('refresh')
+            )
+            ->add(
+                (new SimpleGridAction('common_show_query'))
+                    ->setName($this->trans('Show SQL query', [], 'Admin.Actions'))
+                    ->setIcon('code')
+            )
+            ->add(
+                (new SimpleGridAction('common_export_sql_manager'))
+                    ->setName($this->trans('Export to SQL Manager', [], 'Admin.Actions'))
+                    ->setIcon('storage')
+            );
+    }
+
+    /**
+     * @return RowActionCollection
+     */
+    private function getRowActions(): RowActionCollection
```

Comment:
```suggestion
    protected function getRowActions(): RowActionCollection
```

Can be protected like others methods